File: coldbrew/bots/robot.js
import {BCAbstractRobot, SPECS} from 'battlecode';

var built = false;
var step = -1;

class MyRobot extends BCAbstractRobot {
    turn() {
        step++;

        if (this.me.unit === SPECS.CRUSADER && this.me.team === SPECS.RED) {
            this.log("Preacher health: " + this.me.health);
            return this.attack(-1,-1);
        }

        else if (this.me.unit === SPECS.CASTLE) {
            if (step === 0) {
                this.log("Building a crusader at " + (this.me.x+1) + ", " + (this.me.y+1));
                return this.buildUnit(SPECS.CRUSADER, 1, 1);
            } else this.log("Castle health: " + this.me.health);
        }

    }
}

var robot = new MyRobot();

File: coldbrew/bots/robot.py
from battlecode import BCAbstractRobot, SPECS

__pragma__('iconv')
__pragma__('tconv')
#__pragma__('opov')

class MyRobot(BCAbstractRobot):
    def __init__(self):
        self.step = -1
        super().__init__()

    def turn(self):
        self.step += 1

        if self.me['unit'] == SPECS['CRUSADER'] and self.me['team'] == SPECS['RED']:
            self.log("Crusader health: " + str(self.me['health']))
            return self.attack(-1,-1)

        elif self.me['unit'] == SPECS['CASTLE']:
            if self.step == 0:
                self.log("Building a crusader at " + str(self.me['x']+1) + ", " + str(self.me['y']+1))
                return self.build_unit(SPECS['CRUSADER'], 1, 1)

            else:
                self.log("Castle health: " + str(self.me['health']))

robot = MyRobot()